### DISCUSSION TITLE
Getting error when trying to build retype page with gitpages.

### DISCUSSION THREAD

**Discussion by luislgq** [retypeapp/retype#257] [Q&A] [answered]
Hello, I have followed the instructions found here: https://retype.com/guides/github-actions/#step-1-add-retypeyml-workflow

But I get the following error:

![image](https://user-images.githubusercontent.com/47843020/149971278-f61a3ab0-bb7d-411a-8216-f29f48355deb.png)

here is my workflow file
![image](https://user-images.githubusercontent.com/47843020/149971437-be32418c-cc61-4e2c-937f-6b060ffd1494.png)

I am very much new with all of this, I am unsure of the problem given that it gives no real logs, so not sure how to proceed. It does says the directory does not exist but I do have a .github/workflows in my repo: https://github.com/AtheosGame/AtheosGame.github.io

**Comment by fabriciomurta** [comment]
The problem happens because the workflow file name is `retype.yml`, and you got no `retype.yml` file in your repo with actual Retype configuration. Thus, the action tries to use the workflow file as configuration file...

I see two issues here: 
1. we should actually exclude .github from the search with the GitHub Action
2. we should use another file name as workflow file name in the guide. The guide was written back when retype used other configuration file name, but now it "conflicts" like did for you.

To overcome the issue, simply create your own retype.yml file in the repository root -- which you actually want, as it allows setting up page title and other things you don't want to miss.

Follow [the Quick Start guide](https://retype.com/#quick-start) to run retype locally at your repo, create a default configuration file with `retype watch` and trim it to your needs. Then commit the configuration file and push.

I think we can avoid this issue in the future by addressing the two issues mentioned above, but you can already get up and running with the guide above.

**Comment by luislgq** [comment]
I have changed the name of the workflow file, that did not seem to help. I can follow the quick start but that only works on my local machine. I am trying to set it up on git pages. Is there an example of  a repository using this new configuration? I have no idea how to proceed.

**Reply by DannyBen** [reply]
> Is there an example of a repository using this new configuration?

Maybe one of my repos can provide a hint?

- https://github.com/dannyben/bashly-book
- https://github.com/dannyben/runfile-book

Both are using a retype.yml ([example](https://github.com/DannyBen/bashly-book/blob/master/retype.yml)) and a standard retype build workflow ([example](https://github.com/DannyBen/bashly-book/blob/master/.github/workflows/build.yml)).

**Comment by fabriciomurta** [comment]
oops, I somehow mistook `retype watch` for generating a default config file... Please generate a bare minimum `retype.yml` file with `retype init` command. Commit the file (you probably want to change some info like the page title before commit), and then it should work.

**Comment by geoffreymcgill** [comment, marked as answer]
I submitted a pull request with a few changes. If you merge the pull request, your project should start building correctly. 

https://github.com/AtheosGame/AtheosGame.github.io/pull/3

It looks like there is a bug in the Retype build process where if the root `retype.yml` project configuration file is completely blank, the build fails. We will fix that bug before the next release. 

If you add even one line to your original `retype.yml` file, such as `input: ./`, your project will also build. But, it's probably just easier to accept the pull request as I included a few other minor revisions. 

Hope this helps.

**Reply by luislgq** [reply]
I really really appreciate the help

**Comment by fabriciomurta** [comment]
retypeapp/action-build#21 fixes the issue by no longer searching for configuration files in hidden UNIX directories (directories starting with a dot `.`).

There was also an issue checking the result of the search where the action would try an empty config file if no config file was found. It was fixed as well.

So at least if this scenario repeats in the future (a repo with a `retype.yml` file as workflow file and no `retype.yml` in repo root), Retype will be built using its default configuration. That's what I call a "no-minder" set up. You will only need to worry about the settings after you see your page with default title, etc, and not scary build errors on action logs!

For anyone hitting this issue in the future, currently I believe a best option would be to rely on `retype watch` locally and tune up your page until it looks the way you like before committing the files. The action does run quickly, but that's no match for the instantly refreshed `retype watch` approach.

**Reply by luislgq** [reply]
I really really appreciate the help. I am glad to be a part of this community, although a very new member. hopefully ill be able to help others just as much soon

**Comment by geoffreymcgill** [comment]
The issue above where a blank `retype.yml` file was causing a build failure has been fixed. 

The fix will be included in the next public release of Retype.

Hope this helps.
